source/background.ts

```diff
@@ -68,3 +82,12 @@ browser.runtime.onInstalled.addListener(async ({reason}) => {
 // GitHub Enterprise support
 addToFutureTabs();
 addContextMenu();
+
+// @ts-ignore because this is only needed sometimes
+function featureWasRenamed(from: string, to: string): Migration { // eslint-disable-line @typescript-eslint/no-unused-vars
+	return (options: AnyObject) => {
```

Comment:
Forgot this

It should be:

```suggestion
	return (options: Options) => {
```

but:

```
TS2322: Type '(options: Options) => void' is not assignable to type 'Migration'.
  Types of parameters 'options' and 'savedOptions' are incompatible.
    Type 'Record<string, string | number | boolean>' is missing the following properties from type 'Options': customCSS, personalToken, logging
```

Reply:
Just heading to bed, but I looked at the webex-options-sync types. We can probably update the types for `Migration` to be generic. Maybe [something like this](https://typescript-play.js.org/#code/PQKhCgAIUhBSC2BDADpA9gMwygLgS3QDsBnSJMk3AJ3yIHMz1rIAjddAGwFMlSA6SABUAFt0gBrbgE8yAd3ydOkEUgBu43OkRJcAYxGRcYyJmYJT+bpwAmJAOSQABkSQJuT8rhr5WAV1xuEn4oEGBwbgAPFGZcI2kUcQB5PEJSSABeSAAlbj1mGwAeKloGABpIErp6SAAfSCI-BFZuFnr2Ll4iAD4AbnBwUAhoSAAJPhseFhJ8eldcP2pxMxY5MSJyDajA0jTIPxQbXSCQ6HComOo4ukDqTCQ9cQBZWepdNMKhFIJiMm3uIh2SAAbwAvplIN80iRuiDwABIAAUJHU3BsUN+AC5hBjSBUbNx7n5OLgSNivqlfgBKbFqdD4Gz9UEDG6te6PHIAcVGuLIwIRej8VHQCAAwgBlcXYqoMfrwxLUEjEJCcIToKREaU+WUIzjoej0arYjo8PhMgb5UhxQJUABifiIeh+msgL3ob2dhWy3N5sKyiPQlLxkAJRJJJCpmVh-MgscglqVPH4evoiPs0nQ9ip5oGQA). The only thing I don't like about this is the options can literally be anything then as long as it's an object.

Reply:
I think actually the types of `OptionsSync.Options` should be `typeof defaults`, which is the definition object passed by the user. I don't know if this can be passed around through the instance.

In the [tasks here](https://github.com/bfred-it/webext-options-sync/issues/5) there are more info about what I'm planning to do on that extension as well, I'll mention you when I have that done first.